src/services/game-logic/pieces/King.ts

```diff
@@ -0,0 +1,29 @@
+import { PieceNames } from '../../../enums';
+import { Square } from '../../../models/Square';
+import { Piece } from './Piece';
+
+class King extends Piece {
+	name = PieceNames.KING;
+
+	getPossibleMoves(): Square[] {
+		const positions: Array<number[]> = [
```

Comment:
Jak coś to generyka da się uniknąć też tu: `number[][]` a tak bardziej ogólnie to wcale nie potrzebujesz deklaracji typu, TS się domyśli https://www.typescriptlang.org/docs/handbook/type-inference.html

Reply:
dzięki, poprawione